packages/nimble-components/src/table/specs/README.md

```diff
@@ -1,78 +1,272 @@
+    -   Illustrate how the data is hooked up to the TanStack API
+-   Column Definitions
+    -   Define the interface we will provide for the column providers/components (i.e., width, sorting, allowSort, allowGrouping, etc...)
+    -   Define the base implementation for the column providers that other column providers can extend.
+    -   Define how we intend to support defining a column that uses information from multiple columns (e.g. a hyperlink column that uses data from one column for the URL and the data from another as the text to display)
+        -   What column gets used for sorting?
+        -   Does column API allow for user to supply custom sort function?
+    -   List the set of column providers that Nimble will provide and provide their respective APIs where unique (e.g., formatter for DateTime column)
+    -   Define the table-level APIs for setting the columns
+        -   Slot API on table?
+        -   Alternative/accompanying programmatic API?
+-   Headers
+    -   Define the anatomy of headers in the table DOM
+        -   Require specific component type (i.e. do we need to create a `nimble-table-header`)
+        -   What is the component to use for interaction? Outline Button? Ghost button?
+        -   What and where are the interactive mechanisms/indicators? Sort arrow, etc..
+    -   Can clients customize header content?
+        -   How do we allow for icons to be used in the header (as shown in the visual design doc)
+-   Row Selection
+    -   Define the anatomy of row selection in the table DOM
+        -   Indeterminate checkbox at the far left of each row?
+        -   Selected row CSS/design
+    -   Define events raised when row selection changes/occurs
+    -   Define table-level row-selection API
 -   Grouping
--   Expand/collapse
--   Hierarchical data
+    -   Define interactive mechanism, if any, to provide grouping
+    -   Define table-level API for setting grouping
+    -   Define events raised when grouping changes
+    -   Describe how data hierarchy should work with grouping
+-   Sorting
+    -   Define the table-level API for setting sorting
+    -   Define events raised when sorting changes
+    -   Describe how sorting should with with hierarchical and/or grouped data
+-   Hierarchy
+    -   Define how hierarchy will be represented in the data
+        -   Attribute on table to represent parentId?
+    -   Define event/property APIs needed for dealing with lazily-loaded hierarchical data (possibly out of scope of initial release)
+    -   Describe the UI representation of hierarchical data (there should be a design doc to reference)
+-   Action Menu
+    -   Define how the action menu gets associated with a particular column
+    -   Define the table-level(column-level?) API(s) for applying an action menu to the table (slot, properties, etc...)
+
+_Properties_
+
+Placeholder
```

Comment:
Does "Placeholder" mean that we'll have a future HLD that covers this section? I wonder if it would be helpful to mark each of these future HLDs. e.g. with a 📄 or by adding a link to the corresponding issues once they are created.

In addition to the API section, I think there are some other sections missing from the [custom.component.md](https://github.com/ni/nimble/blob/main/specs/templates/custom-component.md) spec template which may (or may not) also merit their own docs:
- Visual Appearance
- States
- Security
- Tooling
(just did a quick scan, there may be others)

Even if the answer is "N/A" it's good to keep these in the spec to remind reviewers to think them through too.

Reply:
I've added back those sections. The idea is certainly as the other HLDs are defined two things should happen: links to those documents should be added to the Feature APIs section (replacing their current bullet point), and then other sections of this spec should be updated as necessary. This would typically be adding content to the Properties/Events sections, but could also be adding content to the 'Visual Appearance' or 'States' sections.